### PR TITLE
Reorganize code structure

### PR DESCRIPTION
**Pull request by gaearon**
A few changes as a followup to Workspaces.

* Entry points now have natural names, e.g. `React`, `ReactDOM`. Just like before Fiber.
* We now use npm package entry points when importing in the few places we weren't. Such as `ReactFiberReconciler` -> `react-reconciler`.
* Since DOM tests often test both client and server, I removed the mostly arbitary distinction and put most of them together into `packages/dom/__tests__`.
* For DOM implementation, I did another pass at checking which files are client-only/server-only/shared, and separated them into `dom/client`, `dom/server`, and `dom/shared`.
* I deleted a few dead files.
* I moved FB-specific `@preventMunge` directives into flat bundle itself instead of individual source files.
* I separated `TestUtils` and `ReactDOMUnstableNativeDependencies` into a separate bundle type called `RENDERER_UTILS`. The main difference is it shouldn't bundle a reconciler inside itself (unlike a real renderer). This is how I uncovered https://github.com/facebook/react/issues/11280 (which I fixed in https://github.com/facebook/react/pull/11281).

The build products look mostly the same to me except for minor dependency order changes.
Bundle fixture works.

screen shot 2017-10-19 at 6 31 10 pm

Also verified that UFI and Power Editor work in DEV and PROD.

### CODE REVIEW

**Comment by bvaughn** [comment]
> For DOM implementation, I did another pass at checking which files are client-only/server-only/shared, and separated them into dom/client, dom/server, and dom/shared.

Nice!

File: packages/react-native-renderer/src/ReactNativeRenderer.js [hunk 1]
```diff
@@ -41,7 +40,7 @@ ReactFiberErrorLogger.injection.injectDialog(
   ReactNativeFiberErrorDialog.showDialog,
 );
 
-const ReactNativeFiber: ReactNativeType = {
+const ReactNativeRenderer: ReactNativeType = {
```
Comment (bvaughn) [inline packages/react-native-renderer/src/ReactNativeRenderer.js]: It's a bit confusing that we now have `ReactNativeRenderer` that requires a file named `ReactNativeFiberRenderer`.

File: scripts/rollup/bundles.js [hunk 2]
```diff
@@ -23,11 +23,17 @@ const RN_PROD = bundleTypes.RN_PROD;
...
 };
 
+// React
 const ISOMORPHIC = moduleTypes.ISOMORPHIC;
+// Individual renderers. They bundle the reconciler. (e.g. ReactDOM)
 const RENDERER = moduleTypes.RENDERER;
+// Helper packages that access specific renderer's internals. (e.g. TestUtils)
+const RENDERER_UTILS = moduleTypes.RENDERER_UTILS;
+// Standalone reconciler for third-party renderers.
 const RECONCILER = moduleTypes.RECONCILER;
```
Comment (bvaughn) [inline scripts/rollup/bundles.js]: Love it ^

File: scripts/rollup/header.js [hunk 3]
```diff
@@ -9,6 +9,7 @@ function getProvidesHeader(hasteFinalName) {
  *
  * @noflow
  * @providesModule ${hasteFinalName}
+ * @preventMunge
```
Comment (bvaughn) [inline scripts/rollup/header.js]: Nice!

File: packages/react-native-renderer/src/ReactNativeRenderer.js (hunk 1 above)
Comment (gaearon) [reply packages/react-native-renderer/src/ReactNativeRenderer.js]: Agree. I’m also not over the moon with arbitrary ReactDOM/ReactDOMComponent separation. Need to do a pass restructuring those later.

Comment (bvaughn) [reply packages/react-native-renderer/src/ReactNativeRenderer.js]: Yeah, it's def not a blocker to this PR. Just an observation that the naming seems even weirder now 😄

**Comment by gaearon** [comment]
Only prettier failed last time so this is good to go.

File: packages/react-reconciler/index.js [hunk 4]
```diff
@@ -10,6 +10,10 @@
 'use strict';
 
 // TODO: bundle Flow types with the package.
-export type {HostConfig, Deadline, Reconciler} from './src/ReactFiberReconciler';
+export type {
+  HostConfig,
+  Deadline,
+  Reconciler,
+} from './src/ReactFiberReconciler';
```
Comment (Yeti-or) [inline packages/react-reconciler/index.js]: Great song by the way 😆

**Pull request merged**